File: CS/Dashboard_ItemCollectionChanged_Example/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.DataAccess;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Windows.Forms;

namespace Dashboard_ItemCollectionChanged_Example
{
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        public Form1()
        {
            InitializeComponent();
            dashboardDesigner1.CreateRibbon();
            dashboardDesigner1.DashboardLoaded += DashboardDesigner1_DashboardLoaded;
            dashboardDesigner1.CustomizeDashboardTitle += DashboardDesigner1_CustomizeDashboardTitle;
            dashboardDesigner1.Dashboard = new Dashboard();
        }

        private void DashboardDesigner1_DashboardLoaded(object sender, DevExpress.DashboardWin.DashboardLoadedEventArgs e)
        {
            e.Dashboard.ItemCollectionChanged += Dashboard_ItemCollectionChanged;
        }

        private void Dashboard_ItemCollectionChanged(object sender, NotifyingCollectionChangedEventArgs<DashboardItem> e)
        {
            Dashboard dBoard = sender as Dashboard;
            if (e.AddedItems.Count > 0)
            {
                if (e.AddedItems.Count == 1 && dBoard.Items.Count(i => i.Name == e.AddedItems[0].Name) > 1)
                    AddToLog("Duplicated", e.AddedItems);
                else
                    AddToLog("Added", e.AddedItems);
            }
            if (e.RemovedItems.Count > 0)
            {
                AddToLog("Removed", e.RemovedItems);
            }
        }
        private void AddToLog(string actionLabel, IList<DashboardItem> items)
        {
            var strItems = items.Select(i => String.Format("{0} ComponentName= {1}, Name= {2}", actionLabel, i.ComponentName, i.Name));
            memoEdit1.Text += String.Format("{0}\r\n", String.Join("; ", strItems));
            memoEdit1.SelectionStart = memoEdit1.Text.Length - 1;
            memoEdit1.ScrollToCaret();
        }
        private void DashboardDesigner1_CustomizeDashboardTitle(object sender, CustomizeDashboardTitleEventArgs e)
        {
            e.Items.Add(new DashboardToolbarItem("Clear Log", new Action<DashboardToolbarItemClickEventArgs>((args) =>
            {
                memoEdit1.Text = string.Empty;
            }
            )));
            e.Items.Add(new DashboardToolbarItem("Add a new Grid item", new Action<DashboardToolbarItemClickEventArgs>((args) =>
            {
                DashboardDesigner dControl = sender as DashboardDesigner;
                dControl.Dashboard.Items.AddRange(new GridDashboardItem(), new ChartDashboardItem());
            }
            )));
        }
    }
}

File: VB/Dashboard_ItemCollectionChanged_Example/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.DataAccess
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Windows.Forms

Namespace Dashboard_ItemCollectionChanged_Example
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
            dashboardDesigner1.CreateRibbon()
            AddHandler dashboardDesigner1.DashboardLoaded, AddressOf DashboardDesigner1_DashboardLoaded
            AddHandler dashboardDesigner1.CustomizeDashboardTitle, AddressOf DashboardDesigner1_CustomizeDashboardTitle
            dashboardDesigner1.Dashboard = New Dashboard()
        End Sub

        Private Sub DashboardDesigner1_DashboardLoaded(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.DashboardLoadedEventArgs)
            AddHandler e.Dashboard.ItemCollectionChanged, AddressOf Dashboard_ItemCollectionChanged
        End Sub

        Private Sub Dashboard_ItemCollectionChanged(ByVal sender As Object, ByVal e As NotifyingCollectionChangedEventArgs(Of DashboardItem))
            Dim dBoard As Dashboard = TryCast(sender, Dashboard)
            If e.AddedItems.Count > 0 Then
                If e.AddedItems.Count = 1 AndAlso dBoard.Items.Count(Function(i) i.Name = e.AddedItems(0).Name) > 1 Then
                    AddToLog("Duplicated", e.AddedItems)
                Else
                    AddToLog("Added", e.AddedItems)
                End If
            End If
            If e.RemovedItems.Count > 0 Then
                AddToLog("Removed", e.RemovedItems)
            End If
        End Sub
        Private Sub AddToLog(ByVal actionLabel As String, ByVal items As IList(Of DashboardItem))
            Dim strItems = items.Select(Function(i) String.Format("{0} ComponentName= {1}, Name= {2}", actionLabel, i.ComponentName, i.Name))
            memoEdit1.Text += String.Format("{0}" & ControlChars.CrLf, String.Join("; ", strItems))
            memoEdit1.SelectionStart = memoEdit1.Text.Length - 1
            memoEdit1.ScrollToCaret()
        End Sub
        Private Sub DashboardDesigner1_CustomizeDashboardTitle(ByVal sender As Object, ByVal e As CustomizeDashboardTitleEventArgs)
            e.Items.Add(New DashboardToolbarItem("Clear Log", New Action(Of DashboardToolbarItemClickEventArgs)(Sub(args)
                memoEdit1.Text = String.Empty
            End Sub)))
            e.Items.Add(New DashboardToolbarItem("Add a new Grid item", New Action(Of DashboardToolbarItemClickEventArgs)(Sub(args)
                Dim dControl As DashboardDesigner = TryCast(sender, DashboardDesigner)
                dControl.Dashboard.Items.AddRange(New GridDashboardItem(), New ChartDashboardItem())
            End Sub)))
        End Sub
    End Class
End Namespace
